web-app/js/portal/filter/BaseFilterPanel.js

```diff
@@ -101,28 +101,51 @@ Portal.filter.BaseFilterPanel = Ext.extend(Ext.Panel, {
 
 Portal.filter.BaseFilterPanel.newFilterPanelFor = function(cfg) {
 
-    var newFilterPanel;
+    var type = cfg.filter.getType().toLowerCase();
 
-    if (cfg.filter.getType() === "String") {
-        newFilterPanel = new Portal.filter.ComboFilterPanel(cfg);
+    // Todo - DN: Remove when loading filters from DBis removed
+    // Portal types --> GeoServer types (temporary)
+    if (type == "daterange") {
+        type = 'datetime'
     }
-    else if (cfg.filter.getType() == "Date") {
-        newFilterPanel = new Portal.filter.DateFilterPanel(cfg);
+    else if (type === "boundingbox") {
+        type = 'geometrypropertytype'
     }
-    else if (cfg.filter.getType() == "DateRange") {
-        newFilterPanel = new Portal.filter.DateFilterPanel(cfg);
+    else if (type === "number") {
```

Comment:
Yeah, because of line 104 we'll always have a String. It's definitely something to be wary of, though, particularly with numerical comparisons. Another tip is to start strict and loosen it up if necessary. That's essentially what we're doing by using `===`.